pkg/apis/trust/v1alpha1/types_bundle.go

```diff
@@ -158,8 +158,15 @@ type NamespaceSelector struct {
 // in the trust Namespace.
 type SourceObjectKeySelector struct {
 	// Name is the name of the source object in the trust Namespace.
+	// This field must be left empty when `selector` is set
+	//+optional
 	Name string `json:"name"`
 
+	// Selector is the label selector to use to fetch a list of objects. Must not be set
+	// when `Name` is set.
+	//+optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
```

Comment:
I notice that we have created a custom type for the namespace selector in this project API: `NamespaceSelector`. I wonder why? Maybe we want to own the API? @inteon WDYT? I am biased here, but leaning towards using the upstream K8s type.

Reply:
I think it is very valid to embed the metav1 struct here.
Also, the LabelSelector seems to have good comments that can be used to generate useful OpenAPI descriptions.

Reply:
So I guess I can resolve this comment?